src/closures.rs

```diff
@@ -72,7 +72,7 @@ pub(crate) fn rewrite_closure(
 
         result.or_else(|_| {
             // Either we require a block, or tried without and failed.
-            rewrite_closure_block(block, &prefix, context, body_shape)
+            rewrite_closure_block(block, label, &prefix, context, body_shape)
```

Comment:
Instead of passing the `label`, why not just pass the entire `body`, which in this case would be a `ast::ExprKind::Block`? The `ast::ExprKind::Block` rewrite code already handles labels.

Reply:
Because `rewrite_closure_block` does not accept ast::Expr.
However making `rewrite_closure_block` accept ast::Expr with passing is better so I changed in 75c514265cf15ef2cdaccb4be355acb541ea0a28.